internal/tasks/marketplaces-push-disk-images-task.yaml

```diff
@@ -0,0 +1,149 @@
+        }
+        # due to set -e, this catches all EXIT and ERR calls and the task should never fail with nonzero exit code
+        trap 'exitfunc $? $LINENO "$BASH_COMMAND"' EXIT
+
+        # Setup required variables
+        set +x
+        STARMAP_MAPPING=$(params.starmapMapping)
+        STARMAP_MAPPING_FILE=/tmp/starmap.yaml
+        echo "$STARMAP_MAPPING" > "$STARMAP_MAPPING_FILE"
+        set -x
+
+        BASE_DIR="$(mktemp -d)"
+        DISK_IMGS_DIR="${BASE_DIR}/starmap/CLOUD_IMAGES"
+        mkdir -p "${DISK_IMGS_DIR}"
+
+
+        RUNNING_JOBS="\j" # Bash parameter for number of jobs currently running
+        NUM_COMPONENTS=$(jq '.components | length' <<< "$SNAPSHOT_JSON")
+
+        process_component() { # Expected argument is [component json]
+            COMPONENT=$1
+            PULLSPEC=$(jq -er '.containerImage' <<< "${COMPONENT}")
+            IMG_NAME=$(jq -er '.name' <<< "${COMPONENT}")
+            BUILD_NAME=$(jq -er '.contentGateway.productCode' <<< "${COMPONENT}")
+            BUILD_ARCH=$(jq -er '.staged.files[0].filename' <<< "${COMPONENT}")
+            BUILD_ARCH=${BUILD_ARCH%-*}   # Rstrip on -
+            BUILD_ARCH=${BUILD_ARCH##*-}  # Lstrip on -
+            RESOURCES_JSON='{"api": "v1", "resource": "CloudImage", "description": "", "boot_mode": "hybrid", "build": {}, "images": []}'
+            RESOURCES_JSON=$(jq -c --arg build_name "$BUILD_NAME" --arg build_arch "$BUILD_ARCH" \
+                            '.build.name=$build_name|.build.arch=$build_arch' <<< "$RESOURCES_JSON")
+            DESTINATION="${DISK_IMGS_DIR}/${IMG_NAME}"
+            mkdir -p "${DESTINATION}"
+            DOWNLOAD_DIR=$(mktemp -d)
+            cd "$DOWNLOAD_DIR"
+            # oras has very limited support for selecting the right auth entry,
+            # so create a custom auth file with just one entry
+            AUTH_FILE=$(mktemp)
+            select-oci-auth "${PULLSPEC}" > "$AUTH_FILE"
+            oras pull --registry-config "$AUTH_FILE" "$PULLSPEC"
+            NUM_MAPPED_FILES=$(jq '.staged.files | length' <<< "${COMPONENT}")
+            for ((i = 0; i < NUM_MAPPED_FILES; i++)); do
+                FILE=$(jq -c --arg i "$i" '.staged.files[$i|tonumber]' <<< "$COMPONENT")
+                SOURCE=$(jq -er '.source' <<< "$FILE")
+                FILENAME=$(jq -er '.filename' <<< "$FILE")
+                if [ -f "${SOURCE}.gz" ]; then
+                    gzip -d "${SOURCE}.gz"
+                fi
+                if [ -f "${DESTINATION}/${FILENAME}" ]; then
+                    echo -n "Multiple files use the same destination value: $DESTINATION" >&2
+                    echo " and filename value: $FILENAME. Failing..." >&2
+                    exit 1
+                fi
+                if [ "${FILENAME##*\.}" = "vhd" ];
+                    image_type="VHD"
+                else
+                    image_type="AMI"
```

Comment:
I would check the `raw` extension here for AWS/AMI
neither qcow2 nor iso are supported. (In another words; we should not automatically fallback to AMI

Reply:
Good catch!

I've fixed that as well as added a directory for the task with a `README.md` file so it is not compliant with the other tasks.

PTAL again and let me know if you find something else